src/roact.ts

```diff
@@ -0,0 +1,167 @@
+
+            for (const klass of dump.Classes.filter(isCreatableInstance)) {
+                const completionItem = new vscode.CompletionItem(
+                    klass.Name,
+                    vscode.CompletionItemKind.Constant,
+                )
+
+                completionItem.detail = `(class) ${klass.Name}`
+                completionItem.documentation = new vscode.MarkdownString(`[Developer Reference](https://developer.roblox.com/en-us/api-reference/class/${klass.Name})`)
+                completionItems.push(completionItem)
+            }
+
+            return completionItems
+        })
+    }
+
+    public async createCompletionItems(
+        service: ApiClass,
+    ): Promise<vscode.CompletionItem[]> {
+        let completionItems: vscode.CompletionItem[] = []
+
+        for (const member of service.Members) {
+            if (member.MemberType === "Property") {
+                const security = member.Security as ApiPropertySecurity
+                if (security.Read !== "None" && security.Write !== "None") {
+                    continue
+                }
+
+                const completionItem = new vscode.CompletionItem(
+                    member.Name,
+                    vscode.CompletionItemKind.Field,
+                )
+                completionItem.insertText = `${member.Name} = `
+                completionItem.detail = `(property) ${service.Name}.${member.Name}: ${member.ValueType ? member.ValueType.Name : "unknown"}`
+                completionItem.documentation = new vscode.MarkdownString(`[Developer Reference](https://developer.roblox.com/en-us/api-reference/event/${service.Name}/${member.Name})`)
+                completionItems.push(completionItem)
+            } else if (member.MemberType === "Event") {
+                if (member.Security !== "None") {
+                    continue
+                }
+
+                const completionItem = new vscode.CompletionItem(
+                    member.Name,
+                    vscode.CompletionItemKind.Event,
+                )
+                completionItem.insertText = `[Roact.Event.${member.Name}] = `
+                completionItem.detail = `(event) ${service.Name}.${member.Name}(${member.Parameters.map(parameter => `${parameter.Name}: ${parameter.Type ? parameter.Type.Name : "unknown"}`).join(", ")})`
+                completionItem.documentation = new vscode.MarkdownString(`[Developer Reference](https://developer.roblox.com/en-us/api-reference/property/${service.Name}/${member.Name})`)
+                completionItems.push(completionItem)
+            }
+        }
+
+        if (service.Superclass) {
+            const klass = (await this.instances).get(service.Superclass)
+            if (klass) {
+                const inheritedMembers = await this.createCompletionItems(klass)
+                for (const completionItem of inheritedMembers) {
+                    if (completionItem.documentation) {
```

Comment:
```suggestion
                    if (completionItem.documentation !== undefined) {
```

Do this everywhere else you might be doing this.